homework_03/05_Viktoriya_Vasileva/Circles.py

```diff
@@ -0,0 +1,69 @@
+import enum
+import math
+
+class RelativePosition(enum.Enum):
+    NO_COMMON_POINTS = 0
+    TOUCHING = 1
+    INTERSECTING = 2
+    #TOUCHING_INTERNALLY = 3
+    SAME = 3
+
+class Point:
+    def __init__(self, x, y):
+        assert isinstance(x, float) 
+        assert isinstance(y, float)
+        self.x = x
+        self.y = y
+
+    def line(self, point):
+        assert isinstance(point, Point)
+        result = round(math.sqrt( math.pow(point.x - self.x, 2) + math.pow(point.y - self.y, 2)), 5)
+        return result
+        
+class Circle:
+    def __init__(self, point, r):
+        assert isinstance(point, Point)
+        assert isinstance(r, float)
+        assert round(r, 5) > 0
```

Comment:
You do not need to round the radius to the fifth digit in order to be positive. For example, if the radius of the circle is with value 0.0000000000001, after the rounding this will be 0.0 and you will receive Assertion Error. But I think this is a too little mistake here.

Reply:
Teodor is right, you might face with some errors in certain cases.